app/jobs/ingest_archival_media_bag_job.rb

```diff
@@ -0,0 +1,227 @@
+      change_set = DynamicChangeSet.new(ArchivalMediaCollection.new)
+      change_set.validate(source_metadata_identifier: component_id)
+      change_set.sync
+      changeset_persister.save(change_set: change_set)
+    end
+
+    def changeset_persister
+      @changeset_persister ||= PlumChangeSetPersister.new(metadata_adapter: metadata_adapter,
+                                                          storage_adapter: storage_adapter,
+                                                          queue: queue_name)
+    end
+
+    def metadata_adapter
+      Valkyrie::MetadataAdapter.find(:indexing_persister)
+    end
+
+    def storage_adapter
+      Valkyrie::StorageAdapter.find(:disk_via_copy)
+    end
+
+    # get all the data files, group them in file sets (master (original_file), intermediate (for download), access)
+    class ArchivalMediaBagParser
+      attr_reader :path, :audio_files, :file_groups, :component_groups
+      def initialize(path:)
+        @path = path
+        @file_groups = {}
+        @component_groups = {}
+      end
+
+      # group all the files in the bag by barcode_with_part
+      def file_groups
+        return @file_groups unless @file_groups.empty?
+        audio_files.each do |audio_file|
+          @file_groups[audio_file.barcode_with_part] = @file_groups.fetch(audio_file.barcode_with_part, []).append(audio_file)
+        end
+        @file_groups
+      end
+
+      # file_groups in groups by component id
+      def component_groups(component_dict)
+        return @component_groups unless @component_groups.empty?
+        file_groups.keys.each do |barcode_with_part|
+          cid = component_dict.lookup(barcode_with_part)
+          @component_groups[cid] = @component_groups.fetch(cid, []).append(barcode_with_part)
+        end
+        @component_groups
+      end
+
+      private
+
+        # create an AudioPath object for each audio file
+        def audio_files
+          @audio_files ||= path.join("data").each_child.select { |file| [".wav", ".mp3"].include? file.extname }.map { |file| IngestableAudioFile.new(path: file) }
+        end
+    end
+
+    # Decorates the Pathname with some convenience parsing methods
+    # Provides methods needed by FileMetadata.for
+    class IngestableAudioFile
+      attr_reader :path, :barcode_with_part
+      def initialize(path:)
+        @path = path
+      end
+
+      def original_filename
+        path.split.last
+      end
+
+      def mime_type
+        if master? || intermediate?
+          "audio/wav"
+        else
+          "audio/mpeg"
+        end
+      end
+      alias content_type mime_type
+
+      def use
+        if master?
+          Valkyrie::Vocab::PCDMUse.PreservationMasterFile
+        elsif intermediate?
+          Valkyrie::Vocab::PCDMUse.IntermediateFile
+        elsif access?
+          Valkyrie::Vocab::PCDMUse.ServiceFile
+        end
+      end
+
+      def master?
+        path.fnmatch "*_pm.wav"
```

Comment:
I'm not sure if this is important, but I expect `path.to_s.end_with?("_pm.wav")` to be faster.